### ISSUE TITLE
linter.verifyAndFix retrun `fixed: false` but have no messages

### ISSUE THREAD

**Issue by xtx1130** [eslint/eslint#11281]
**Tell us about your environment**

* **ESLint Version: 5.10.0**
* **Node Version: v9.10.0**
* **npm Version: 5.6.0**

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint
**Please show your full configuration:**
https://github.com/xtx1130/icode-style/blob/master/src/rules.json

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
msg = linter.verifyAndFix('/**\n * @file index.js\n */\nimport RollAnimation from \'./roll-animation.vue\';\n\nRollAnimation.install = function (Vue) {\n    Vue.component(RollAnimation.name, RollAnimation);\n};\n\nexport default RollAnimation;\n', {
      rules, // I am using https://github.com/xtx1130/icode-style/blob/master/src/rules.json
      parser: 'babel-eslint'
    })
```

**What did you expect to happen?**
```js
{
  fixed: 'false',
  messages: [] // the messages shouldn't be empty
}
```

**What actually happened? Please include the actual, raw output from ESLint.**
```js
{
  fixed: 'false',
  messages: [] // the messages are empty
}
```
**Are you willing to submit a pull request to fix this bug?**
I am not sure if this is a bug or not, but if The code won't be fixed, shouldn't it will tell me what's wrong?

**Comment by xtx1130** [comment]
@platinumazure ,thanks for you explanation.
> but if your code is lint-free, then the array should be empty.

Can I think the `fixed` key is just a status about if the code has changed or not, if the code doesn't need to fix, the `fixed` is false and the `messages` will be an empty array?

>Do you see errors if you use CLIEngine instead of Linter? 

There is no error when using CLI.
